### PR TITLE
[Merged by Bors] - feat: Linter that checks that Prop classes are Props

### PR DESCRIPTION
**Pull request by fpvandoorn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/Tactic/Tauto.lean [hunk 1]
```diff
@@ -124,6 +124,7 @@ def distribNot : TacticM Unit := withMainContext do
   liftMetaTactic' (distribNotAux fvars)
 
 /-- Config for the `tauto` tactic. Currently empty. TODO: add `closer` option. -/
+@[nolint structureInType]
```
Comment (digama0) [inline Mathlib/Tactic/Tauto.lean:127]: I think structures with no fields should be explicitly excluded from the lint. We basically don't have enough information about how the structure is intended to be used to make an accurate assessment about whether it should be a Prop.

Comment (fpvandoorn) [reply Mathlib/Tactic/Tauto.lean:127]: that's fair

**Comment by fpvandoorn** [comment]
That was more painful than I thought.

bors merge

**Pull request closed without merging**